OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/ReachableMethodAnalysis.scala

```diff
@@ -125,12 +109,28 @@ trait ReachableMethodAnalysis extends FPCFAnalysis with TypeConsumerAnalysis {
 
     private[this] def continuationForCallers(
         oldCallers: Callers,
-        tacEP:      EPS[Method, TACAI]
+        tacEPOpt:   Option[EPS[Method, TACAI]]
     )(
         update: SomeEPS
     ): ProperPropertyComputationResult = {
         val newCallers = update.asInstanceOf[EPS[DeclaredMethod, Callers]]
-        processMethod(newCallers, oldCallers, tacEP)
+        processMethod(newCallers, oldCallers, tacEPOpt)
     }
+}
 
+trait DefinedBodyReachableMethodAnalysis extends ReachableMethodAnalysis {
```

Comment:
Since there seems to be only one ReachableMethodAnalysis that does process methods without body, might it be more sensible to switch the logic around, with the base trait not doing that and the specialized trait doing it?

Reply:
I pondered this and did could not answer this question. On one hand, if the specialized trait is used far more than the base trait, something might be wrong. Additionally, the specialized trait right now restricts the abilities of the base trait (if you want to view "does not process but just returns no result" as a restriction) which feels a bit against the core principles of inheritance.

On the other hand, even if the base trait does not handle unknown bodies, it still needs to define extension points for
1. The unknown body specialised trait to handle such unknown bodies
2. A `processMethod` or similar that has an `EOptionP` as a parameter.

In short (using the flipped version): If you only extend the `ReachableMethodAnalysis` you should only need to implement the `EPS` version of `processMethod` since you know you have a body. If you extend the `UnknownBodyReachableMethodAnalysis`, you should only need to implement the `EOptionP` version of `processMethod` which can handle both scenarios.

All in all, I feel more comfortable with the current solution, but I am happy to replace it with a better one if you have some ideas.